components/dialog/index.css

```diff
@@ -34,337 +34,321 @@ governing permissions and limitations under the License.
-    margin-block-start: var(--spectrum-dialog-vertical-offset);
-
-    &.is-open {
-      transform: translateY(0);
-    }
-  }
+  outline: none; /* Hide focus outline around header */
 }
 
-/* Responsive Dialogs will fill the screen on small viewports */
-@media
-only screen and (max-device-width: 400px),
-only screen and (max-device-height: 350px) {
-  .spectrum-Dialog--responsive {
-    inline-size: 100%;
-    block-size: 100%;
-    max-inline-size: 100%;
-    max-block-size: 100%;
-    border-radius: 0;
-  }
-  .spectrum-Dialog-wrapper {
-    .spectrum-Dialog--responsive {
-      margin-block-start: 0; /* Prevent the vertical offset bit */
-    }
-  }
+.spectrum-Dialog-typeIcon {
+  grid-area: typeIcon;
 }
 
-.spectrum-Dialog--noDivider {
-  .spectrum-Dialog-header {
-    &::after {
-      display: none;
-    }
-  }
+.spectrum-Dialog .spectrum-Dialog-divider {
+  grid-area: divider;
+  width: 100%;
+  margin-top: var(--spectrum-dialog-rule-margin-top);
+  margin-bottom: var(--spectrum-dialog-rule-margin-bottom);
 }
 
-.spectrum-Dialog--dismissible {
-  .spectrum-Dialog-footer {
-    /* You cannot have buttons in a dismissible dialog */
+.spectrum-Dialog--noDivider {
+  .spectrum-Dialog-divider {
     display: none;
   }
-
-  .spectrum-Dialog-closeButton {
-    display: initial;
+  .spectrum-Dialog-heading {
+    padding-bottom: calc(var(--spectrum-dialog-rule-margin-top, var(--spectrum-global-dimension-static-size-150)) + var(--spectrum-dialog-rule-margin-bottom, var(--spectrum-global-dimension-static-size-200)) + var(--spectrum-dialog-rule-height, var(--spectrum-global-dimension-static-size-25)));
   }
 }
 
-.spectrum-Dialog-closeButton {
-  position: absolute;
-  inset-block-start: calc(-1 * var(--spectrum-global-dimension-size-350));
-  inset-inline-end: calc(-1 * var(--spectrum-global-dimension-size-350));
-}
-
-.spectrum-Dialog-typeIcon {
-  display: block;
-  margin-inline-start: var(--spectrum-dialog-icon-margin-left);
-  flex-shrink: 0;
-  align-self: flex-start;
-}
-
 .spectrum-Dialog-content {
-  display: block;
+  grid-area: content;
   box-sizing: border-box;
 
+  /*
+  v2 is currently worse, inputs that get focus rings get a slight clipping unless overflow: visible replaces this line
+  we should think of a better way to handle this, see padding/margin below for fix
+  */
   overflow-y: auto;
   -webkit-overflow-scrolling: touch;
 
-  flex: 1 1 auto;
-
   outline: none; /* Hide focus outline */
 
   font-size: var(--spectrum-dialog-content-text-size);
   font-weight: var(--spectrum-dialog-content-text-font-weight);
   line-height: var(--spectrum-dialog-content-text-line-height);
-}
 
-@media screen and (-ms-high-contrast: active), (-ms-high-contrast: none) {
-  .spectrum-Dialog-content {
-    /* Temporary IE 11 fix */
-    max-block-size: 70vh;
-  }
+  /* this is kinda dumb, but needed for the keyboard focus rings so they don't get clipped. is there a better way to treat this */
```

Comment:
not sure if there is a better way. space needs to be made to ensure the focus ring is visible. 
only other option I would see requires scripting and placing buttons via `position:absolute` on top of the dialog. not the best solution IMO.

Reply:
Negative margin is fine here, it's hacky, but that's the entire web.

Reply:
@garthdb Would prefer that these globals are assigned to a diff variable though.

Reply:
Good call.